packages/core/src/dialects/sqlite/query-interface.js

```diff
@@ -369,4 +369,29 @@ export class SqliteQueryInterface extends AbstractQueryInterface {
 
     return `${prefix}_${suffix}`;
   }
+
+  /**
+   * @override
+   */
+  async bulkDelete(table, where, options, model) {
+    const bulkDeleteOptions = { ...options, limit: null, type: QueryTypes.BULKDELETE };
+
+    const sql = bulkDeleteOptions.truncate === true
+      ? this.queryGenerator.truncateTableQuery(table, bulkDeleteOptions)
+      : this.queryGenerator.deleteQuery(table, where, bulkDeleteOptions, model);
+
+    // unlike bind, replacements are handled by QueryGenerator, not QueryRaw
+    delete bulkDeleteOptions.replacements;
+
+    const results = [];
+    const subQueries = sql.split(';').filter(q => q !== '');
+
+    for (const subQuery of subQueries) {
+      const res = await this.sequelize.query(subQuery, bulkDeleteOptions);
+      results.push(res);
+    }
+
+    // Return the number of affected rows which is the first element of the result array
+    return results[0];
```

Comment:
We'll need to run multiple queries more often in sqlite (e.g. the uniqueKeys option in createTableQuery should be updated to generate multiple queries) so it would be good to extract this to to a separate function

Reply:
I've added a function to the abstract query interface as it might be useful to some other dialects as well.